source/common/event/dispatcher_impl.cc

```diff
@@ -157,7 +157,9 @@ DispatcherImpl::createClientConnection(Network::Address::InstanceConstSharedPtr
 
 Network::DnsResolverSharedPtr DispatcherImpl::createDnsResolver(
     const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
-    const bool use_tcp_for_dns_lookups) {
+    const bool use_tcp_for_dns_lookups,
+    const envoy::config::core::v3::AreaDnsLookupOptionFlags& area_dns_lookup_option_flags) {
+  //    bool no_default_search_domain_for_dns_lookups) {
```

Comment:
I will remove these commented unused code. you can ignore them

Reply:
See comment above about incorporating the `use_tcp_for_dns_lookups` parameter into the flags.